loleaflet/src/core/Socket.js

```diff
@@ -268,6 +268,30 @@ L.Socket = L.Class.extend({
 			     'background:#ddf;color:black', color, 'color:black');
 	},
 
+	// The problem: if we process one websocket message at a time, the
+	// browser -loves- to trigger a re-render as we hit the main-loop,
+	// this takes ~200ms on a large screen, and worse we get
+	// producer/consumer issues that can fill a multi-second long
+	// buffer of web-socket messages in the client that we can't
+	// process so - slurp and the emit at idle - its faster to delay!
+	_slurpMessage: function(e) {
+		var that = this;
+		if (!this._slurpQueue || !this._slurpQueue.length) {
+			setTimeout(function() {
+				console.log2('Slurp events ' + that._slurpQueue.length);
```

Comment:
I understand we expect that._slurpQueue to always be set and valid, but experience shows that we better check before accessing.

Reply:
we check it before accessing surely?

Reply:
is `console.log2` correct? isn't that a typo?